lib/ansible/modules/notification/mail.py

```diff
@@ -241,9 +269,46 @@ def main():
     secure_state = False
     sender_phrase, sender_addr = parseaddr(sender)
 
+    pgp_recipients = module.params.get('pgp_recipients')
+    pgp = pgp_recipients or module.params.get('pgp')
+
     if not body:
         body = subject
 
+    if pgp:
+        try:
+            import gnupg
+        except ImportError:
+            module.fail_json(rc=1, msg='OpenPGP encryption requires python-gnupg')
+
+        # TODO: Implement proper PGP/MIME
+        if attach_files:
+            module.fail_json(rc=1, msg='OpenPGP encryption does not yet support attachments')
+
+        gpg = gnupg.GPG()
+        gpg.encoding = charset
+
+        if pgp_recipients:
+            import re
+            from ansible.module_utils.six import string_types
+
+            def is_invalid(s):
```

Comment:
Since 2.4.0 custom type validation (#27183) is available. If you use it, be aware of #31445.

Reply:
#31445 isn't yet useable, though, right?

Reply:
Yes, the feature is available but there is a bug.